icefoganalytics/travel-authorization
# Part 2: Wire Travel Desk Request Tab

Relates to:

- Part of solving https://github.com/icefoganalytics/travel-authorization/issues/167
- [TravelDeskWireFrame.pdf](https://github.com/icefoganalytics/travel-authorization/files/14016343/TravelDeskWireFrame.pdf)
- [UserStories-TravelDesk.docx](https://github.com/icefoganalytics/travel-authorization/files/14016344/UserStories-TravelDesk.docx)
- [TravelDeskDataModel (1).xlsx](https://github.com/icefoganalytics/travel-authorization/files/14016345/TravelDeskDataModel.1.xlsx)
- https://github.com/ytgov/travel-authorization/tree/main/web/src/modules/travelDesk
- https://github.com/icefoganalytics/travel-authorization/pull/182

# Context

Using the existing Travel Desk forms (https://github.com/ytgov/travel-authorization/tree/main/web/src/modules/travelDesk) rewire the travel desk request to exist in the travel desk tab of the travel auth.

Travel Desk forms should be created anytime a user is traveling by air.

![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/74adc941-50cd-4219-a723-062822c4a94a)

# Implementation

- Standardized the travel desk flight requests schema and improved CRUD operations.
- Refactored flight request dialogs into separate components for better modularity.
- Enhanced responsiveness for mobile devices and fixed UI bugs like date formatting and extra borders.
- Streamlined data handling and component structure, including script setup conversion and code linting.
- Fixed critical bugs related to data display and event handling.
- Introduced auto-save features across various components to enhance user convenience.

# Screenshots

Flight request table
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/40e83fcd-534b-4109-8f54-76698b2788cf)

Add flight dialog
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/ca813612-c89b-41af-bf3c-42d791a0411d)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to the "My Travel Requests" page via the top drop down nav.
5. Create a travel authorization.
6. In the details section, select at least one travel method with "aircraft" type.
7. Generate the travel estimate.
8. Submit the travel authorization to yourself as the supervisor.
9. Go to Manger View page accessible via the drop down nav.
10. Click on the new request from the "Pending Approvals" panel in the top left.
11. Scroll to the bottom and approve the request.
12. Go back to the "My Travel Requests" page, and find your new travel request in the table.
13. Click on the Submit Travel Desk Request button to go to the Request tab. (NEW: it works now!)
14. Check that the Flight Request section load correctly, showing a blank table, and the "additional information" section.
15. Add a flight. Check that the table gets populated.
16. Edit the flight via the pencil icon. Check that the table refreshes showing the updated information.
17. Delete the flight. Note that there is a new - absolutely minimal - confirmation dialog for deletion.
18. Check that you can add additional information for the flight, and that it auto-saves correctly.

Note that the Rental car and below do not work yet.
